From 5aee0639c0177073dd548b3c4ce316f0bb278849 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 30 Jan 2020 19:16:05 +0000
Subject: [PATCH] Update for qtdeclarative 5.15

* adapt to changes from https://code.qt.io/cgit/qt/qtdeclarative.git/commit/?h=5.15&id=cc1a604c704f848927b3fa0a97b0a50b0b79d2a4
* better described in:
  https://www.qt.io/blog/qml-type-registration-in-qt-5.15
* fixes:
  qemux86-64 luna-next[347]: WARNING: 10:56:30.818: file:///usr/palm/luna-next/shells/card/qml/CardShell.qml:22:1: plugin cannot be loaded for module "LuneOS.Components": Cannot protect module LuneOS.Components 1 as it was never registered


Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 modules/LuneOS/Application/plugin/luneoswindow.h | 3 +++
 modules/LuneOS/Application/plugin/plugin.cpp     | 2 +-
 modules/LuneOS/Components/plugin/plugin.cpp      | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/LuneOS/Application/plugin/luneoswindow.h b/modules/LuneOS/Application/plugin/luneoswindow.h
index d33becc..33bd3c7 100644
--- a/modules/LuneOS/Application/plugin/luneoswindow.h
+++ b/modules/LuneOS/Application/plugin/luneoswindow.h
@@ -28,6 +28,7 @@ class LuneOSWindow : public QQuickApplicationWindow
     Q_PROPERTY(unsigned int parentWindowId READ parentWindowId WRITE setParentWindowId NOTIFY parentWindowIdChanged)
     Q_PROPERTY(bool keepAlive READ keepAlive WRITE setKeepAlive NOTIFY keepAliveChanged)
     Q_PROPERTY(bool loadingAnimationDisabled READ loadingAnimationDisabled WRITE setLoadingAnimationDisabled NOTIFY loadingAnimationDisabledChanged)
+    QML_ELEMENT
 
 public:
     explicit LuneOSWindow();
@@ -89,5 +90,7 @@ private:
     void configure();
     void reset();
 };
+    
+QML_DECLARE_TYPEINFO(LuneOSWindow, QML_HAS_ATTACHED_PROPERTIES)
 
 #endif // APPLICATIONWINDOW_H
diff --git a/modules/LuneOS/Application/plugin/plugin.cpp b/modules/LuneOS/Application/plugin/plugin.cpp
index ef31ef0..a52b22e 100644
--- a/modules/LuneOS/Application/plugin/plugin.cpp
+++ b/modules/LuneOS/Application/plugin/plugin.cpp
@@ -28,7 +28,7 @@ Plugin::Plugin(QObject *parent) :
 void Plugin::registerTypes(const char *uri)
 {
     Q_ASSERT(uri == QLatin1String("LuneOS.Application"));
-    qmlRegisterType<LuneOSWindow>(uri, 1, 0, "LuneOSWindow");
+    qmlRegisterTypesAndRevisions<LuneOSWindow>(uri, 1);
 }
 
 void Plugin::initializeEngine(QQmlEngine *engine, const char *uri)
diff --git a/modules/LuneOS/Components/plugin/plugin.pro b/modules/LuneOS/Components/plugin/plugin.pro
index c22db63..f868022 100644
--- a/modules/LuneOS/Components/plugin/plugin.pro
+++ b/modules/LuneOS/Components/plugin/plugin.pro
@@ -4,6 +4,10 @@ TARGET = ../LuneOSComponents
 QT += core-private qml qml-private quick quick-private
 CONFIG += qt plugin
 
+CONFIG += qmltypes
+QML_IMPORT_NAME = "LuneOS.Components"
+QML_IMPORT_MAJOR_VERSION = "1"
+
 TARGET = $$qtLibraryTarget($$TARGET)
 uri = LuneOS.Components
 
